x-pack/plugins/observability_solution/infra/public/components/ml/anomaly_detection/job_setup_screen.tsx

```diff
@@ -125,22 +150,36 @@ export const JobSetupScreen = (props: Props) => {
     indices,
     partitionField,
     startDate,
+    telemetry,
+    filter,
+    props.jobType,
   ]);
 
   const onFilterChange = useCallback(
     (f: string) => {
       setFilter(f || '');
       setFilterQuery(convertKueryToElasticSearchQuery(f, metricsView?.dataViewReference) || '');
+      telemetry.reportAnomalyDetectionFilterFieldChange({
+        job_type: props.jobType,
+        filter_field: f,
+      });
     },
-    [metricsView?.dataViewReference]
+    [metricsView?.dataViewReference, telemetry, props.jobType]
   );
 
   /* eslint-disable-next-line react-hooks/exhaustive-deps */
   const debouncedOnFilterChange = useCallback(debounce(onFilterChange, 500), [onFilterChange]);
 
-  const onPartitionFieldChange = useCallback((value: Array<{ label: string }>) => {
-    setPartitionField(value.map((v) => v.label));
-  }, []);
+  const onPartitionFieldChange = useCallback(
+    (value: Array<{ label: string }>) => {
+      setPartitionField(value.map((v) => v.label));
+      telemetry.reportAnomalyDetectionPartitionFieldChange({
+        job_type: props.jobType,
+        partition_field: value.length > 0 ? value[0].label : null,
```

Comment:
Q: Shouldn't the `partition_field` accept an array in this case? If there is more than 1 label, this will report a incorrect info, wouldn't it?

Reply:
I checked again because I thought looking at the code that would be possible, but the UI doesn't let you select more than one, so it's all of them or just one field. 
But yeah, it doesn't work as it is (without selecting one), I'll fix it
Thanks for spotting this